app/javascript/packs/banner_editor.jsx

```diff
@@ -97,6 +99,285 @@ export default class BannerEditor extends React.Component {
     Spinner.show('spinner', 'spinner-container');
   }
 
+  changeState(state){
+    switch(state){
+      case 'editor':
+        this.setState({state: 'editor'})
+        document.getElementsByClassName("brave-rewards-banner-container")[0].style.width = "1200px"
+        break;
+      case 'editor-logo-set':
+        this.setState({state: 'editor-logo-set'})
+        document.getElementsByClassName("brave-rewards-banner-container")[0].style.width = "600px"
+        document.getElementsByClassName("brave-rewards-banner-container")[0].style.height = "676px"
+        break;
+      case 'editor-logo-not-set':
+        this.setState({state: 'editor-logo-not-set'})
+        document.getElementsByClassName("brave-rewards-banner-container")[0].style.width = "600px"
+        document.getElementsByClassName("brave-rewards-banner-container")[0].style.height = "676px"
+        break;
+      case 'editor-logo-added':
+        this.setState({state: 'editor-logo-added'})
+        break;
+    }
+  }
+
+  renderControlBar(){
```

Comment:
It makes more sense to me to extract these `renderX()` functions out into several components as these seem isolated. A large react file is hard for me to read 🙈

Reply:
Agreed Cory, see the proposed refactor plan here #1289, it includes separate components 👍